lib/msf/core/windows_version.rb

```diff
@@ -0,0 +1,160 @@
+# -*- coding: binary -*-
+module Msf
+  #
+  # Represents the version of a Windows operating system
+  #
+  class WindowsVersion
+
+    VER_NT_WORKSTATION = 1
+    VER_NT_DOMAIN_CONTROLLER = 2
+    VER_NT_SERVER = 3
+
+    XP_SP0 = Rex::Version.new('5.1.2600.0')
```

Comment:
Is there a reason we don't have version mappings for Windows pre-XP? Specifically Windows 2000. If not, I can dig up the versions for SP0 to SP4.